realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -753,6 +753,12 @@ private Table getPrimaryKeyTable() {
         if (sharedRealm == null) {
             return null;
         }
+
+        // FIXME: The PK table creation should be handle by Object Store after integration of OS Schema.
+        if (!sharedRealm.hasTable(PRIMARY_KEY_TABLE_NAME)) {
```

Comment:
The PK table may still get created from here in some cases, eg.:`moreFieldsChangedThanLocalRefTableSize` test. So I would like to keep the old logic in this PR and clean it up in the future.